docs/data/upload.md

```diff
@@ -0,0 +1,113 @@
+       filepath=os.path.join(my_data_directory, file_name)
+     )
+   ```
+
+4. Save and run your code once.
+   Running your code more than once will duplicate the data.
+   View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+## Sync data with the `sync` method
+
+The Viam Golang SDK includes a data service API with a [`sync`](/data/#sync) method that syncs all the data from your machine to the cloud.
+The [cloud sync tools](/data/cloud-sync/) sync data automatically according to how you configure cloud sync, but if you prefer to sync programatically, you can instead run a script that calls the `sync` method.
+
+The following steps assume you already have a [data capture service configured](/data/capture/#add-the-data-management-service).
+If you want to use the `sync` API method as the only way to sync data, you should toggle sync off in your data service config (`"sync_disabled": true` in raw JSON).
+
+1. Install the [Viam Go SDK](https://github.com/viamrobotics/rdk/tree/main/robot/client).
+
+2. Create a Go script file in a directory of your choice.
+   Go to your machine's **Code sample** tab in the [Viam app](https://app.viam.com) and click **Golang**.
+   Copy the code from that tab into your script file to establish a connection from your computer to your machine's instance in the Viam app.
+
+3. Add the required import to the list of imports at the top of your script:
+
+   ```go {class="line-numbers linkable-line-numbers"}
+   import (
+     "go.viam.com/rdk/services/datamanager"
+   )
+   ```
+
+4. In your `main()` function (or in a function called from `main`), call [the `sync` method as documented](/data/#sync).
+   For example:
+
+   ```go {class="line-numbers linkable-line-numbers"}
+   // Get the data management service from your machine
+   // Use the name of the data service you configured on your machine in place of "my_data_service"
+   data, err := datamanager.FromRobot(robot, "my_data_service")
+
+   // Sync data stored on the machine to the cloud.
+   err := data.Sync(context.Background(), nil)
+   ```
+
+5. Save, compile, and run your code.
+   View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+## Upload images with the Viam mobile app
+
+Upload images as machine data straight from your phone, skipping the normal data capture and cloud synchronization process, through the [Viam mobile app](/fleet/#the-viam-mobile-app).
+This is useful if you want to capture images for training machine learning models on the go.
+
+1. Select **Home** on the mobile app, choose your organization, and select a location and machine.
+2. Click the menu button marked "**...**" in the upper right corner.
+3. Click **Upload Images**.
+4. Select each image you want to upload, and click **Add**.
+
+With regard to metadata, these images are associated with the machine part you selected, and aren't associated with a component or method.
```

Comment:
Nit: tiny clarity.

```suggestion
With regard to metadata, these images are associated with the machine part you selected, and aren't associated with a specific component or method.
```

Q: is `method` correct here? I know of upload data being associated with components, but not sure of method here.

Reply:
That is a great question. I didn't write this section (just pulled it over from the other page). I know that you configure specific methods in the data capture config on a given component (see "Add method" button in the app) but not sure if that is associated with the data after it's captured... @sguequierre do you know?